packages/amplication-client/src/Layout/PageLayout.tsx

```diff
@@ -1,29 +1,48 @@
-import { TabItem, Tabs } from "@amplication/ui/design-system";
+import { Tabs } from "@amplication/ui/design-system";
 import classNames from "classnames";
 import React, { useCallback } from "react";
-import { LockedFeatureIndicator } from "../Components/LockedFeatureIndicator";
 import "./PageLayout.scss";
+import { FeatureTabItem } from "../Resource/ResourceHome";
+import {
+  EntitlementType,
+  FeatureControlContainer,
+} from "../Components/FeatureControlContainer";
+import { FeatureIndicator } from "../Components/FeatureIndicator";
 
 type Props = {
   children: React.ReactNode;
   className?: string;
   pageTitle?: string;
-  tabs?: TabItem[];
+  tabs?: FeatureTabItem[];
 };
 
 const CLASS_NAME = "amp-page-layout";
 
 function PageLayout({ children, className, tabs }: Props) {
   const tabsComponent = useCallback(() => {
-    return tabs.map((tab, index) => (
-      <Tabs.Tab
-        key={index}
-        {...tab}
-        lockedFeatureIndicator={
-          tab.disabled && <LockedFeatureIndicator featureName={tab.name} />
-        }
-      />
-    ));
+    return tabs.map((tab, index) => {
+      return tab.license ? (
+        <FeatureControlContainer
+          key={index}
+          featureId={tab.license}
+          entitlementType={EntitlementType.Boolean}
+          reversePosition={true}
+          render={({ icon, disabled }) => (
```

Comment:
this is bad practice' we need to handle the children

Reply:
I explained why I need these two options. can you explain why this is a bad practice?

Reply:
I answered to the first comment, lets take it offline